os/fs/driver/mtd/smart.c

```diff
@@ -5166,6 +5408,92 @@ static int smart_journal_erase(FAR struct smart_struct_s *dev, uint16_t physsect
 	return ret;
 }
 
+/****************************************************************************
+ * Name: smart_journal_check_log_empty
+ *
+ * Description:
+ *   Return false if log is not empty, return true if all log datas are empty.
+ *
+ ****************************************************************************/
+
+static bool smart_journal_check_log_empty(journal_log_t *log)
+{
+	size_t log_size = sizeof(journal_log_t);
+	uint8_t *buf = (uint8_t *)log;
+
+	for (int i = 0; i < log_size; i++) {
+		if (buf[i] != CONFIG_SMARTFS_ERASEDSTATE) {
+			return false;
+		}
+	}
+	return true;
+}
+
+/****************************************************************************
+ * Name: smart_journal_dump_log
+ *
+ * Description:
+ *   Dump all log of journal
+ *
+ ****************************************************************************/
+
+static void smart_journal_dump_log(FAR struct smart_struct_s *dev)
+{
+	uint32_t addr;
+	uint32_t count = 0;
+	uint32_t area0 = dev->neraseblocks * dev->sectorsPerBlk * dev->sectorsize;
+	uint32_t area1 = area0 + (dev->njournaleraseblocks / 2) * dev->sectorsPerBlk * dev->sectorsize;
+	uint32_t last = dev->geo.neraseblocks * dev->sectorsPerBlk * dev->sectorsize;
+	journal_log_t log;
+	int ret;
+
+	/* TODO All dump log will be hide Later, Now enable it to develop journaling */
+	addr = area0;
+
+	while (addr < last) {
+		if (addr == area0 || addr == area1) {
+			uint8_t id_bits;
+			ret = MTD_READ(dev->mtd, addr, 1, &id_bits);
+			if (ret < 0) {
+				fdbg("read failed! ret : %d addr : %u\n", addr);
+				break;
+			}
+			fdbg("area #%d status : 0x%x\n", (count == 0)? 0 : 1, id_bits);
+			addr++;
+		} else {
+			ret = MTD_READ(dev->mtd, addr, sizeof(journal_log_t), (FAR uint8_t *)&log);
+			if (ret < 0) {
+				fdbg("read failed! ret : %d addr : %u\n", addr);
+				break;
+			}
+			
+			if (smart_journal_check_log_empty(&log)) {
+				fdbg("end of journal\n");
+				break;
+			}
+
+			fdbg("Journal data #%d\n", count);
```

Comment:
can we use different debug log for error and info....fdbg for error and fvdbg for info?

Reply:
see above comment. This should be kept to debug journaling.